languages/elixir/exercises/concept/enum/.docs/after.md

```diff
@@ -0,0 +1,132 @@
+`Enum` is a very useful module that provides a set of algorithms for working with enumerables. It offers:
+
+- sorting ([`sort/2`][enum-sort/2], [`sort_by/2`][enum-sort_by/2], [`reverse/2`][enum-reverse/2], [`shuffle/1`][enum-shuffle/1]),
+- filtering ([`filter/2`][enum-filter/2], [`reject/2`][enum-reject/2], [`uniq/1`][enum-uniq/1], [`uniq_by/2`][enum-uniq_by/2]),
+- grouping ([`chunk_by/2`][enum-chunk_by/2], [`chunk_every/2`][enum-chunk_every/2], [`chunk_while/4`][enum-chunk_while/4], [`group_by/3`][enum-group_by/3]),
+- counting ([`count/2`][enum-count/2], [`frequencies/1`][enum-frequencies/1], [`frequencies_by/2`][enum-frequencies_by/2])
+- searching ([`find/3`][enum-find/3], [`find_index/2`][enum-find_index/2], [`find_value/3`][enum-find_value/3]),
+- finding min/max values ([`min/3`][enum-min/3], [`max/3`][enum-max/3], [`min_by/4`][enum-min_by/4], [`max_by/4`][enum-max_by/4], [`min_max/2`][enum-min_max/2], [`min_max_by/3`][enum-min_max_by/3]),
+- reducing ([`reduce/3`][enum-reduce/3], [`reduce_while/3`][enum-reduce_while/3], [`sum/1`][enum-sum/1]),
+- and much more.
+
+## Enumerable
+
+In Elixir, an enumerable is any data type that implements the `Enumerable` protocol. Those are:
+
+- [`List`][list]
+- [`Map`][map]
+- [`Range`][range]
+- [`Stream`][stream]
+- [`MapSet`][mapset]
+- [`Function`][function]
+- [`Date.Range`][data-range]
+- [`IO.Stream`][io-stream]
+- [`File.Stream`][file-stream]
+
+Don't worry if you don't know them all yet.
+
+As we will learn in a later exercise about protocols, anyone can implement the `Enumerable` protocol for their own custom data structure.
+
+## Passing anonymous functions as arguments
+
+Many `Enum` functions accept a function as a second argument. The expected return value of the passed function differs depending on the use case. It could be...
+
+A boolean value:
+
+```elixir
+Enum.reject([1, 2, 3, 4, 5], fn x -> rem(x, 2) == 0 end)
+# => [1, 3, 5]
+```
+
+A new element:
+
+```elixir
+Enum.map([1, 2, 3, 4, 5], fn x -> x + 10 end)
+# => [11, 12, 13, 14, 15]
+```
+
+A special tuple:
+
+```elixir
+Enum.reduce_while([1, 2, 3, 4, 5], 0, fn x, acc ->
```

Comment:
`reduce_while` is introduced before introducing `reduce` below. This may be a bit confusing.

Reply:
The example is gone now.